pkg/terraform/workspace_test.go

```diff
@@ -0,0 +1,454 @@
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	k8sExec "k8s.io/utils/exec"
+	testingexec "k8s.io/utils/exec/testing"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/crossplane-contrib/terrajet/pkg/resource/json"
+	tferrors "github.com/crossplane-contrib/terrajet/pkg/terraform/errors"
+)
+
+var (
+	testType              = "very-cool-type"
+	applyType             = "apply"
+	lineage               = "very-cool-lineage"
+	terraformVersion      = "1.0.10"
+	version               = 1
+	serial                = 3
+	directory             = "./testdata"
+	changeSummaryAdd      = `{"@level":"info","@message":"Plan: 1 to add, 0 to change, 0 to destroy.","@module":"terraform.ui","@timestamp":"0000-00-00T00:00:00.000000+03:00","changes":{"add":1,"change":0,"remove":0,"operation":"plan"},"type":"change_summary"}`
+	changeSummaryUpdate   = `{"@level":"info","@message":"Plan: 0 to add, 1 to change, 0 to destroy.","@module":"terraform.ui","@timestamp":"0000-00-00T00:00:00.000000+03:00","changes":{"add":0,"change":1,"remove":0,"operation":"plan"},"type":"change_summary"}`
+	changeSummaryNoAction = `{"@level":"info","@message":"Plan: 0 to add, 0 to change, 0 to destroy.","@module":"terraform.ui","@timestamp":"0000-00-00T00:00:00.000000+03:00","changes":{"add":0,"change":0,"remove":0,"operation":"plan"},"type":"change_summary"}`
+
+	state = &json.StateV4{
+		Version:          uint64(version),
+		TerraformVersion: terraformVersion,
+		Serial:           uint64(serial),
+		Lineage:          lineage,
+		RootOutputs:      map[string]json.OutputStateV4{},
+		Resources:        []json.ResourceStateV4{},
+	}
+)
+
+func newFailureWorkspace() *Workspace {
+	return NewWorkspace(directory, WithExec(&testingexec.FakeExec{
+		CommandScript: []testingexec.FakeCommandAction{
+			func(_ string, _ ...string) k8sExec.Cmd {
+				b := bytes.Buffer{}
+				b.WriteString(errBoom.Error())
+
+				return &testingexec.FakeCmd{
+					CombinedOutputScript: []testingexec.FakeAction{
+						func() ([]byte, []byte, error) {
+							return b.Bytes(), nil, errBoom
+						},
+					},
+				}
+			},
+		},
+	},
+	), WithLogger(newLogger()))
+}
+
+func newWorkspaceWithChangeSummary(changeSummary string) *Workspace {
+	return NewWorkspace(
+		directory, WithExec(&testingexec.FakeExec{
+			CommandScript: []testingexec.FakeCommandAction{
+				func(_ string, _ ...string) k8sExec.Cmd {
+					b := bytes.Buffer{}
+					b.WriteString(changeSummary)
+
+					return &testingexec.FakeCmd{
+						CombinedOutputScript: []testingexec.FakeAction{
+							func() ([]byte, []byte, error) {
+								return b.Bytes(), nil, nil
+							},
+						},
+					}
+				},
+			},
+		},
+		), WithLogger(newLogger()))
+}
+
+func newLogger() logging.Logger {
+	zl := zap.New(zap.UseDevMode(true))
+	return logging.NewLogrLogger(zl.WithName("provider-jet-test"))
+}
+
+func timeNow() *time.Time {
+	t := time.Unix(0, 0)
+	return &t
+}
```

Comment:
```suggestion
```

Having `var now = time.Now()` and referencing it via `&now` could be less code.

Reply:
Done!